modules/s3-bucket/main.tf

```diff
@@ -1,17 +1,26 @@
 locals {
   enabled = module.this.enabled
 
-  aws_partition = data.aws_partition.current.partition
+  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : data.template_file.bucket_policy.rendered
+}
+
+module "custom_policy_account_roles" {
+  count = local.enabled && var.custom_policy_enabled ? 1 : 0
+
+  source = "../account-map/modules/roles-to-principals"
 
-  custom_policy_account_arns = [
+  iam_role_arn_template = "arn:aws:iam::%s:role/%s-%s-%s-%s"
+  role_map = {
     for acct in var.custom_policy_account_names :
-    format("arn:%s:iam::%s:root", local.aws_partition, module.account_map.outputs.full_account_map[acct])
-  ]
+    acct => ["root"]
+  }
 
-  bucket_policy = var.custom_policy_enabled ? data.aws_iam_policy_document.custom_policy[0].json : data.template_file.bucket_policy.rendered
-}
+  root_account_stage_name = var.account_map_stage_name
+  global_environment_name = var.account_map_environment_name
+  global_tenant_name      = coalesce(var.account_map_tenant_name, module.this.tenant)
```

Comment:
We should not need these. Those values should be pre-configured in `account-map`.
```suggestion
```